documentation/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.md

```diff
@@ -0,0 +1,587 @@
+
+### 2. Add user preferences admin view
+
+[See "Moderator preferences" for rationale](#moderator-preferences)
+
+- Sub-class the `User` model to create `OpenverseUser` and add a new binary JSON
+  column, `preferences`
+- Create a new admin view for `OpenverseUser` called `User preferences` that has
+  only a single checkbox control to turn on and off the `moderator.blur_images`
+  setting
+
+### 3. Table view improvements
+
+[See "Table view improvements" for rationale](#table-view-improvements)
+
+- Create a new view based on the media object default model view but sort media
+  by pending report count and the oldest report's age
+- Exclude media items that have no reports
+- Filter to only show media with pending reports by default, but allow showing
+  all reported media
+
+### 4. `ModerationDecision`
+
+[See "`ModerationDecision`" for rationale](#moderationdecision)
+
+- Create the new `ModerationDecision` models
+  - Create an `AbstractModerationDecision` for each media type
+- Add new `decision_id` column to report models
+- Remove bulk report updates from the report models' `save` methods
+
+### 5. Media moderation view
+
+[See "Media moderation view" for rationale](#media-moderation-view)
+
+This is in three separate issues:
+
+1. Create a new media view to replace the existing view
+
+- Use the basic model admin but with only "view" settings for all model fields
+- We'll need to expand the rendered fields to include things like the rendered
+  (interactive) thumbnail and audio player
+- It should display (and not allow editing) the following:
+  - title
+  - description
+  - tags
+  - creator with link to creator landing page on source (if applicable)
+  - source/provider
+  - the work's thumbnail itself, blurred if the moderator has not turned off
+    blurring
+    - clicking on the thumbnail should un-blur the work
+  - if an audio work, show the basic HTML5 audio player
+  - links to the foreign landing URL and Openverse.org page of the work
+  - whether sensitive text detection matched on the work
+    - check if the work is in the filtered index, if not, then it has sensitive
+      text
+- Show chronolical lists of all moderation decisions and reports for the work
+  - Do this using an
+    [`TabularInline`](https://docs.djangoproject.com/en/4.2/ref/contrib/admin/#inlinemodeladmin-objects)
+    for each
+
+2. If the user viewing the page has content moderation permissions, add the
+   relevant actions for the work based on it
+
+- See the
+  [relevant overview section for described actions](#media-moderation-view) and
+  the conditions for when to show them
+- Add an free text form field "explanation" to collect for the moderation
+  decision
+- For any action taken, we only need to update/save a single report with the new
+  status. The `AbstractMediaReport` class already handles updating any other
+  pending reports in it's `save` method without creating duplicate/conflicting
+  mature or deleted media objects.
```

Comment:
This sentence should be deleted, right? Now that we are removing that logic from the AbstractMediaReport, and instead updating each selected report individually.

Reply:
Oops, yes, definitely.